pkg/landscaper/controllers/deployitem/controller.go

```diff
@@ -15,55 +16,50 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	lscore "github.com/gardener/landscaper/apis/core"
 	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
 	lsv1alpha1helper "github.com/gardener/landscaper/apis/core/v1alpha1/helper"
 )
 
 const (
-	PickupTimeoutReason    = "PickupTimeout"    // for error messages
-	PickupTimeoutOperation = "WaitingForPickup" // for error messages
+	PickupTimeoutReason      = "PickupTimeout"    // for error messages
+	PickupTimeoutOperation   = "WaitingForPickup" // for error messages
+	AbortingTimeoutReason    = "AbortingTimeout"  // for error messages
+	AbortingTimeoutOperation = "WaitingForAbort"  // for error messages
 )
 
 // NewController creates a new deploy item controller that handles timeouts
 // To detect pickup timeouts (when a DeployItem resource is not reconciled by any deployer within a specified timeframe), the controller checks for a timestamp annotation.
 // It is expected that deployers remove the timestamp annotation from deploy items during reconciliation. If the timestamp annotation exists and is older than a specified duration,
 // the controller marks the deploy item as failed.
-// rawPickupTimeout is a string containing the pickup timeout duration, either as 'none' or as a duration that can be parsed by time.ParseDuration.
-func NewController(log logr.Logger, c client.Client, scheme *runtime.Scheme, rawPickupTimeout string) (reconcile.Reconciler, error) {
+// pickupTimeout is a string containing the pickup timeout duration, either as 'none' or as a duration that can be parsed by time.ParseDuration.
+func NewController(log logr.Logger, c client.Client, scheme *runtime.Scheme, pickupTimeout, abortingTimeout, defaultTimeout lscore.Duration) (reconcile.Reconciler, error) {
```

Comment:
the code is totally fine but I think the dereference in the addController method is strange. and could lead to strange nil pointer exceptions

```suggestion
func NewController(log logr.Logger, c client.Client, scheme *runtime.Scheme, pickupTimeout, abortingTimeout, defaultTimeout *lscore.Duration) (reconcile.Reconciler, error) {
```